beetsplug/autobpm.py

```diff
@@ -11,81 +11,71 @@
 # The above copyright notice and this permission notice shall be
 # included in all copies or substantial portions of the Software.
 
-"""Uses Librosa to calculate the `bpm` field.
-"""
+"""Uses Librosa to calculate the `bpm` field."""
 
+from __future__ import annotations
 
-from librosa import beat, load
-from soundfile import LibsndfileError
+from typing import Iterable
 
-from beets import ui, util
+import librosa
+
+from beets.importer import ImportTask
+from beets.library import Item, Library
 from beets.plugins import BeetsPlugin
+from beets.ui import Subcommand, should_write
 
 
 class AutoBPMPlugin(BeetsPlugin):
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
         self.config.add(
             {
                 "auto": True,
                 "overwrite": False,
+                "beat_track_kwargs": {},
             }
         )
 
-        if self.config["auto"].get(bool):
+        if self.config["auto"]:
             self.import_stages = [self.imported]
 
-    def commands(self):
-        cmd = ui.Subcommand(
+    def commands(self) -> list[Subcommand]:
+        cmd = Subcommand(
             "autobpm", help="detect and add bpm from audio using Librosa"
         )
         cmd.func = self.command
         return [cmd]
 
-    def command(self, lib, opts, args):
-        self.calculate_bpm(lib.items(ui.decargs(args)), write=ui.should_write())
+    def command(self, lib: Library, _, args: list[str]) -> None:
```

Comment:
I agree that the unused arguments should be marked, but I think their names should be preserved, i.e. using `_opts` instead of just `_`.  It's just a bit clearer, especially since the whole codebase isn't well-typed yet.  Also, a type annotation should still be used for the argument.

Reply:
I agree regarding general clarity, but do you reckon this is needed here given that this argument is unused? I find that `_` in such case clearly communicates that. I find the presence of the type a bit confusing if the argument is not used.

Note that my opinion here is mostly based on refactoring interfaces like the one that `BeetsPlugin` provides: in this case using `_` notation allows the interface (and the type) to change without the need to update the code here.

Essentially, I'm applying the [ISP principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) to the function arguments here

> Within [object-oriented design](https://en.wikipedia.org/wiki/Object-oriented_design), [interfaces](https://en.wikipedia.org/wiki/Interface_(object-oriented_programming)) provide layers of abstraction that simplify code and create a barrier preventing coupling to [dependencies](https://en.wikipedia.org/wiki/Dependency_inversion_principle). A system may become so coupled at multiple levels that it is no longer possible to make a change in one place without necessitating many additional changes.[[1]](https://en.wikipedia.org/wiki/Interface_segregation_principle#cite_note-ASD-1) Using an interface or an [abstract class](https://en.wikipedia.org/wiki/Abstract_class) can prevent this side effect.

Reply:
That makes sense, I'm just not used to it.  Any preference, @Serene-Arc?